src/main/scala/io/iohk/ethereum/blockchain/sync/FastSyncNodesRequestHandler.scala

```diff
@@ -76,49 +79,56 @@ class FastSyncNodesRequestHandler(
   }
 
   private def handleMptNode(mptNode: MptNode): Seq[HashType] = mptNode match {
-    case n: MptLeaf =>
-      val evm = n.getAccount.codeHash
-      val storage = n.getAccount.storageRoot
+    case n: LeafNode =>
+      import AccountImplicits._
+      //if this fails it means that we have leaf node which is part of MPT that do not stores account
+      //we verify if node is paert of the tree by checking its hash before we call handleMptNode() in line 44
+      val account = Try(n.value.toArray[Byte].toAccount)
```

Comment:
minor readability issue (it was hard for me to get intention here). Imo pattern match on try would be clearer, without all these transformations. Something like:
```  
case Success(acc) => Some(acc)
case Failure(e) =>
    log.debug(s"Leaf node without account, error while trying to decode account ${e.getMessage}")
    None
```